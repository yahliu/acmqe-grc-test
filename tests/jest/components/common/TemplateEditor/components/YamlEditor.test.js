/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import YamlEditor from '../../../../../../src-web/components/common/TemplateEditor/components/YamlEditor'
import renderer from 'react-test-renderer'

describe('YamlEditor', () => {
  it('renders as expected', () => {
    const props = {
      height: '41vh',
      onYamlChange() {},
      readOnly: true,
      setEditor() {},
      width: '50vw',
      yaml: 'YamlEditorTestYaml1'
    }
    const component = renderer.create(
      <YamlEditor  {...props} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders as expected', () => {
    const props = {
    }
    const component = renderer.create(
      <YamlEditor  {...props} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
