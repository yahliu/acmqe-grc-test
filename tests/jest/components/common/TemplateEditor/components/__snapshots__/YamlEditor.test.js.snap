// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YamlEditor renders as expected 1`] = `
<div
  className="yamlEditorContainer"
  style={
    Object {
      "display": "block",
      "minHeight": "100px",
    }
  }
>
  <div
    className="react-monaco-editor-container"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`YamlEditor renders as expected 2`] = `
<div
  className="yamlEditorContainer"
  style={
    Object {
      "display": "block",
      "minHeight": "100px",
    }
  }
>
  <div
    className="react-monaco-editor-container"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;
